railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -130,6 +130,9 @@ def config_when_updating
       assets_config_exist            = File.exist?("config/initializers/assets.rb")
       csp_config_exist               = File.exist?("config/initializers/content_security_policy.rb")
 
+      load_defaults = File.read("config/application.rb").match(/config\.load_defaults (\d\.\d)/)
+      @config_target_version = load_defaults[1] if load_defaults
```

Comment:
If there's no `load_defaults` in there, but we're upgrading... maybe this should be:

```ruby
@config_target_version = load_defaults[1] || "5.0"
```

Reply:
If there's no `config.load_defaults` line in the file, and we regenerate it with `config.load_defaults 5.0`, we've opted the app into [a set of defaults](https://github.com/rails/rails/blob/936a96a98b543c593590064356cb04e884b875b7/railties/lib/rails/application/configuration.rb#L66-L77) that they weren't getting before - which is what we're trying to avoid here.

I think we should skip the `load_defaults` line in the template if we didn't find it in the current file.

Reply:
Yeah, it's a fair point. But since I would like to recommend using `load_defaults`, want to avoid skip.
So, I added version `4.2` which does not affect existing configs and used it there.

@matthewd What do you think?

Reply:
I think omitting it for upgrades is fine... people should really be doing a stepwise upgrade anyway, so it will be rare for 5.2 to see an app that was never upgraded to 5.0 or 5.1. (That's also why I went for 5.0... it's not correct, but seemed less bother than making the whole line optional just for a corner case.)

Offering 4.2 seems more likely to be confusing, because it's not really true... IIRC we changed a number of the 'real'/baseline defaults in 5.0?

Reply:
My concern was `belongs_to_required_by_default`. It effects to the whole application. 
But I fixed it to `5.0` in order to agree that `4.2` is confusing and that it seems to have responded somewhat already with the update from 5.0 to 5.1. 
(I also do not agree much to make the whole line an option for that corner case)